### ISSUE TITLE
Lag on iOS when sending small pieces of data quickly

### ISSUE THREAD

**Issue by christomitov** [MobileChromeApps/mobile-chrome-apps#522]
Okay so I'm writing an app that sends touch drag events as mouse movements to PC. I'm throttling the requests so that its manageable whilst retaining performance and this seems to work flawlessly on Android. However on iOS the mouse becomes increasingly more laggy, almost like the packets start getting buffered more and more before being sent. 

Originally I thought this was because the ability to turn of Nagle's algorithm was missing from the iOS implementation. But I've commented out setNoDelay on the Android side and still have great performance there with no slow down. 

I know this is difficult to debug without actually seeing the code but I'm just curious off hand if you guys know what low level implementations in GCDAsyncSocket differ from the java implementation that may contribute to this anomaly.

**Comment by christomitov** [comment]
Okay just a little more insight. I started profiling my app when I was receiving Memory Errors in Xcode console. Below are some screenshots:

http://i.imgur.com/xN9lzAV.png  - Note JS Garbage Collector is increasing in size after every tcp event
http://i.imgur.com/Ya1aMj1.png - The app caps out at about 140 mb on my test device, at this point tcp is unusable

I've profiled the application with DDMS and the android version seems to have no leaks at all. I think I can deduce that the iOS version of the plugin is the culprit? I will run some more profiling tests to see what leaks I can find in my code but so far there's nothing.
